File: Algorithms/Searching-Algorithms/linear-search/linearSearch.py
def linear_search(arr, target):
    
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1


def main():
    n = int(input("Enter the number of elements: "))

    arr = []
    print(f"Enter {n} elements:")
    for _ in range(n):
        arr.append(int(input()))

    target = int(input("Enter the target element: "))

    result = linear_search(arr, target)

    if result == -1:
        print("Element not found in the array.")
    else:
        print(f"Element {target} found at index {result}.")


if __name__ == "__main__":
    main()

File: Algorithms/Searching-Algorithms/linear-search/linearSearch.java
import java.util.Scanner;
import java.util.Arrays;

public class linearSearch {

    // Linear search method
    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i; // return index of the target
            }
        }
        return -1; // target not found
    }

    // Main method
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter the number of elements:");
        int n = scanner.nextInt();

        int[] arr = new int[n];

        System.out.println("Enter " + n + " elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        System.out.println("Enter the target element:");
        int target = scanner.nextInt();

        // Perform linear search
        int result = linearSearch(arr, target);

        if (result == -1) {
            System.out.println("Element not found in the array.");
        } else {
            System.out.println("Element " + target + " found at index " + result + ".");
        }

        scanner.close();
    }
}
